### PR TITLE
Added authentication example

### PR DESCRIPTION
**Pull request by impronunciable**
This example show how to add authentication to a next app using passwordless + email

### CODE REVIEW

**Comment by impronunciable** [comment]
#153

File: examples/with-auth/server.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  server.use(session({ secret: 'Meow', resave: false, saveUninitialized: true }))
+  server.use(auth.sessionSupport())
+  server.use(auth.acceptToken({ successRedirect: '/' }))
+
+  server.post('/sendtoken', auth.requestToken((email, delivery, fn) =>
+  fn(null, email)), (req, res) => res.json('ok'))
+
+  server.get('/me', (req, res) => res.json(req.user || null))
+
+  server.post('/logout', (req, res) => req.session.destroy(() => res.json('ok')))
```
Comment (rauchg) [inline examples/with-auth/server.js]: I would really like to move this into a separate server. I don't want to promote bloated servers that collocate the "backend" and the "middle-end"

Comment (arunoda) [inline examples/with-auth/server.js]: +1 for that.

File: examples/with-auth/pages/private.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { authenticate, logout, logoutEvent } from '../utils'
+
+export default class Private extends React.Component {
+  static async getInitialProps ({ req, res }) {
+    const user = await authenticate(req, res)
+    return { user }
+  }
+
+  componentDidMount () {
+    console.log('asd')
```
Comment (jpsc) [inline examples/with-auth/pages/private.js]: Unnecessary log 🙂

File: examples/with-auth/server.js (hunk 1 above)
Comment (impronunciable) [inline examples/with-auth/server.js]: Will we expose this in the example or will be a separate black-box service?

Comment (sedubois) [inline examples/with-auth/server.js]: Maybe use Auth0 (as I do so far in my proto) then people can start from there if they want to use their own stuff?

File: examples/with-auth/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+
+# Passwordless Authentication example
+
+## How to use
+
+Download the example (or clone the repo)[https://github.com/zeit/next.js.git]:
```
Comment (sedubois) [inline examples/with-auth/README.md]: Link should be: `[or clone the repo](https://github.com/zeit/next.js)` 😄

File: examples/with-auth/server.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+const express = require('express')
+const session = require('express-session')
+const bodyParser = require('body-parser')
+const next = require('next')
+const auth = require('./auth')
+
+const dev = process.env.NODE_ENV !== 'production'
+const app = next({ dir: '.', dev })
```
Comment (sedubois) [inline examples/with-auth/server.js]: no need for `dir: '.'`

Comment (timneutkens) [inline examples/with-auth/server.js]: It's actually taken from the `custom-server-with-express` example. I guess for verbosity of the example.

Comment (timneutkens) [inline examples/with-auth/server.js]: You're obviously right that it defaults to `'.'` anyway.

File: examples/with-auth/README.md (hunk 3 above)
Comment (timneutkens) [inline examples/with-auth/README.md]: Fixed 👍

File: examples/with-auth/auth.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+const passwordless = require('passwordless')
+const MemoryStore = require('passwordless-memorystore')
+const email = require('emailjs')
+const config = require('./config.json')
+
+const smtpServer = email.server.connect(config.emailServer)
```
Comment (sedubois) [inline examples/with-auth/auth.js]: There is no `emailServer` in `config.json`. What should it be?

**Comment by impronunciable** [comment]
Closing in favor of #1141

**Comment by timneutkens** [comment]
@impronunciable ❤️

**Comment by camstuart** [comment]
This example seems to have vanished?

**Comment by tharakabimal** [comment]
@impronunciable @arunoda  Yeah, it is nowhere to be found.

**Comment by tomasdev** [comment]
Brrrrrring it back @impronunciable <3

**Pull request closed without merging**